File: VB/SaveDocument/Form1.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
'#Region "#usings"
Imports DevExpress.XtraRichEdit
Imports DevExpress.XtraRichEdit.Export
Imports DevExpress.XtraRichEdit.Services

'#End Region  ' #usings
Namespace SaveDocumentSample

    Partial Public Class Form1
        Inherits DevExpress.XtraBars.Ribbon.RibbonForm

        '#Region "#savecommandaddservice"
        Public Sub New()
            InitializeComponent()
            Dim commandFactory As CustomRichEditCommandFactoryService = New CustomRichEditCommandFactoryService(richEditControl1, richEditControl1.GetService(Of IRichEditCommandFactoryService)())
            richEditControl1.ReplaceService(Of IRichEditCommandFactoryService)(commandFactory)
        End Sub

        '#End Region  ' #savecommandaddservice
        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
            richEditControl1.LoadDocument("TextWithImages.rtf", DocumentFormat.Rtf)
        End Sub

        '#Region "#save"
        Private Sub btnSaveToFile_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnSaveToFile.Click
            richEditControl1.SaveDocument("SavedDocument.rtf", DocumentFormat.Rtf)
            Dim fi As IO.FileInfo = New IO.FileInfo("SavedDocument.rtf")
            Dim msg As String = String.Format("The size of the file is {0:#,#} bytes.", fi.Length.ToString("#,#"))
            XtraMessageBox.Show(msg)
        End Sub

        Private Sub richEditControl1_BeforeExport(ByVal sender As Object, ByVal e As BeforeExportEventArgs) Handles richEditControl1.BeforeExport
            If e.DocumentFormat = DocumentFormat.Rtf Then
                Dim reduceFileSize As DialogResult = XtraMessageBox.Show("Do you want to reduce the file size and avoid exporting the document theme and properties?",
                                                                         "Warning", MessageBoxButtons.YesNo, MessageBoxIcon.Warning)
                Dim options As RtfDocumentExporterOptions = TryCast(e.Options, RtfDocumentExporterOptions)
                If options IsNot Nothing Then
                    Select Case reduceFileSize
                        Case DialogResult.Yes
                            options.ExportTheme = False
                            options.ExportedDocumentProperties = DocumentPropertyNames.None
                        Case DialogResult.No
                            options.ExportTheme = True
                            options.ExportedDocumentProperties = DocumentPropertyNames.All
                    End Select
                End If
            End If
        End Sub
        '#End Region  ' #save
    End Class
End Namespace

File: CS/SaveDocument/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;
#region #usings
using DevExpress.XtraRichEdit;
using DevExpress.XtraRichEdit.Export;
using DevExpress.XtraRichEdit.Services;
#endregion #usings

namespace SaveDocumentSample
{
    public partial class Form1 : DevExpress.XtraBars.Ribbon.RibbonForm
    {
        #region #savecommandaddservice
        public Form1()
        {
            InitializeComponent();

            CustomRichEditCommandFactoryService commandFactory =
                new CustomRichEditCommandFactoryService(richEditControl1,
                    richEditControl1.GetService<IRichEditCommandFactoryService>());
            richEditControl1.ReplaceService<IRichEditCommandFactoryService>(commandFactory);
        }
        #endregion #savecommandaddservice

        private void Form1_Load(object sender, EventArgs e)
        {
            richEditControl1.LoadDocument("TextWithImages.rtf", DocumentFormat.Rtf);
        }
        #region #save
        private void btnSaveToFile_Click(object sender, EventArgs e)
        {
            richEditControl1.SaveDocument("SavedDocument.rtf", DocumentFormat.Rtf);
            System.IO.FileInfo fi = new System.IO.FileInfo("SavedDocument.rtf");
            string msg = String.Format("The size of the file is {0:#,#} bytes.", fi.Length.ToString("#,#"));
            XtraMessageBox.Show(msg);
        }

        private void richEditControl1_BeforeExport(object sender, DevExpress.XtraRichEdit.BeforeExportEventArgs e)
        {
            if (e.DocumentFormat == DocumentFormat.Rtf)
            {
                DialogResult reduceFileSize = XtraMessageBox.Show("Do you want to reduce the file size and avoid exporting the document theme and properties?",
                "Warning", MessageBoxButtons.YesNo, MessageBoxIcon.Warning);

                RtfDocumentExporterOptions options = e.Options as RtfDocumentExporterOptions;
                if (options != null)
                {
                    switch (reduceFileSize)
                    {
                        case DialogResult.Yes:
                            options.ExportTheme = false;
                            options.ExportedDocumentProperties = DocumentPropertyNames.None;
                            break;
                        case DialogResult.No:
                            options.ExportTheme = true;
                            options.ExportedDocumentProperties = DocumentPropertyNames.All;
                            break;
                    }
                }
            }
        }
        #endregion  #save
    }
}